src/sentry/api/endpoints/internal_beacon.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from django.conf import settings
+from rest_framework.response import Response
+
+from sentry import get_version, options
+from sentry.api.base import Endpoint
+from sentry.http import safe_urlopen
+from sentry.tasks.beacon import BEACON_URL
+
+logger = logging.getLogger("beacon")
+
+
+class InternalBeaconEndpoint(Endpoint):
+    permission_classes = ()
+
+    def post(self, request):
+        error = None
+        install_id = options.get("sentry:install-id")
+
+        if not settings.SENTRY_BEACON:
+            logger.info(
+                "beacon_metric.skipped", extra={"install_id": install_id, "reason": "disabled"}
+            )
+            return Response(status=204)
+
+        if settings.DEBUG:
+            logger.info(
+                "beacon_metric.skipped", extra={"install_id": install_id, "reason": "debug"}
+            )
+            return Response(status=204)
+
+        anonymous = options.get("beacon.anonymous") is not False
+
+        payload = {
+            "type": "metric",
+            "install_id": install_id,
+            "version": get_version(),
+            "anonymous": anonymous,
+        }
+
+        if not anonymous:
+            payload["admin_email"] = options.get("system.admin-email")
```

Comment:
Maybe this part can be abstracted out into a common beacon helper as it is the same with the beacon pinger?

Reply:
I think this logic is simple enough that it's fine to be duplicated, if anything we could just remove it completely as we don't need admin email (and we can reference it via install_id anyway)